File: ZezeJava/src/Zeze/Util/Benchmark.java
package Zeze.Util;

import java.lang.management.ManagementFactory;
import com.sun.management.OperatingSystemMXBean;

public class Benchmark {
    private long startTime;
    private long startProcessCpuTime;
    private OperatingSystemMXBean os;
    public Benchmark() {
        os = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
        startProcessCpuTime = os.getProcessCpuTime();
        startTime = System.nanoTime();
    }

    public void Report(String name, long tasks) {
        double cpu =  (os.getProcessCpuTime() - startProcessCpuTime) / 1_000_000_000;
        var endTime = System.nanoTime();
        var elapsedTime = endTime - startTime;
        var seconds = (double)elapsedTime / 1_000_000_000;
        var cpupercent = cpu / seconds;
        System.out.println(String.format(
                "%s tasks/s=%.2f time=%.2fs cpu=%.2fs concurrent=%.2f",
                name, (tasks / seconds), seconds, cpu, cpupercent));
    }
}

File: Zeze/Util/Benchmark.cs
using System;
using System.Diagnostics;
using System.Runtime.Versioning;

namespace Zeze.Util
{
    public class Benchmark
    {
        private readonly long startTime;
        private PerformanceCounter counter;

        [SupportedOSPlatform("windows")]
        private void StartPerformanceCounter()
        {
            counter = new PerformanceCounter(
                "Processor",
                "% Processor Time",
                "_Total",
                true);
        }

        public Benchmark()
        {
            if (OperatingSystem.IsWindows())
                StartPerformanceCounter();

            startTime = DateTime.Now.Ticks;
        }

        [SupportedOSPlatform("windows")]
        private float GetProcessCpuTime()
        {
            return counter.NextValue();
        }

        public void Report(string name, long tasks)
        {
            var cpu = OperatingSystem.IsWindows() ? GetProcessCpuTime() : 0.0f;
            var endTime = DateTime.Now.Ticks;
            var elapsedTime = endTime - startTime;
            var seconds = (double)elapsedTime / TimeSpan.TicksPerMillisecond / 1000;
            var concurrent = cpu / seconds;

            // format
            var stasks = string.Format("{0:0.00}", tasks / seconds);
            var stime = string.Format("{0:0.00}", seconds);
            var scpu = string.Format("{0:0.00}", cpu);
            var sconcurrent = string.Format("{0:0.00}", concurrent);
            Console.WriteLine($"{name} tasks/s={stasks} time={stime} cpu={scpu} concurrent={sconcurrent}");
        }
    }
}
